packages/client/src/components/interface/DownloadButton.tsx

```diff
@@ -238,18 +237,9 @@ function DownloadButtonComponent(props: DownloadButtonProps & HOCProps) {
   }, [downloadConfigs, client, downloadDeclaration])
   const hideModal = useCallback(() => setAssignModal(null), [])
   const unassign = useCallback(async () => {
-    if (assignment) {
-      unassignDeclaration(compositionId, client)
-    } else {
-      deleteDeclaration(compositionId, client)
```

Comment:
Do we know why the deletion was there earlier? 

Also, do we know why the assignment is empty in some cases and it doesn't go the to the unassignDeclaration like expected?

Reply:
I can now see a probable reason why this was here, the `unassignDeclaration` does a network request to unassign it, but if it's already unassigned from you, the network request will error. Then the downloaded record isn't deleted. 

So the delete is for a use case where you have downloaded the record, but assignment is with someone else. Then it just deletes it from you without a network request.

Reply:
https://github.com/opencrvs/opencrvs-core/blob/a3c4fab35794d32b2a13fc54515aab09a8f71120/packages/workflow/src/records/handler/unassign.ts#L54
We could return `200 OK` or `204 No Content` here instead of throwing an error when it's already been unassigned.